### PR TITLE
Merge predix extensions 75.6.0 and predix_prunes/jwt_grant_75.6.0 with dockerized UAA pipeline fixes into rc_75.6.0

### PR DESCRIPTION
**Pull request by akarollil**
This is for merging Dockerize pipeline fixes from the predix_extensions_75.6.0 branch (see https://github.com/GESoftware-CF/uaa/pull/347) and the predix_prunes/jwt_grant_75.6.0 branches (https://github.com/GESoftware-CF/uaa/commit/034223931b97a65c6e62834efa74ee2e8e9edd4b) into rc_75.6.0

### CODE REVIEW

**Comment by akarollil** [comment]
Build with JWT Bearer Grant Acceptance test passing for Dockerize UAA pipeline: https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA/job/Dockerize/job/PR-354/1/

**Comment by akarollil** [comment]
Build with JWT Bearer Grant Acceptance test passing for Dockerize UAA pipeline: https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA/job/Dockerize/job/PR-354/2/

File: docker.gradle [hunk 1]
```diff
@@ -92,34 +91,35 @@ task copyContainerFiles(type: Copy) {
...
-         "ENFORCE_CLIENT_ASSERTION_HEADER=" + System.getenv('ENFORCE_CLIENT_ASSERTION_HEADER'),
-         "KEY_PROVIDER_SERVICE_URL=" + System.getenv('KEY_PROVIDER_SERVICE_URL'),
-         "KEY_PROVIDER_TOKEN_URL=" + System.getenv('KEY_PROVIDER_TOKEN_URL'),
-         "KEY_PROVIDER_CLIENT_ID=" + System.getenv('KEY_PROVIDER_CLIENT_ID'),
-         "KEY_PROVIDER_CLIENT_SECRET=" + System.getenv('KEY_PROVIDER_CLIENT_SECRET'),
-         "KEY_PROVIDER_DEFAULT_INSTANCE=" + System.getenv('KEY_PROVIDER_DEFAULT_INSTANCE')]
+  hostConfig.portBindings = ['8080:8080']
+  envVars.putAll([
+          'CATALINA_OPTS': '-Dspring.profiles.active=default',
+          'ENFORCE_CLIENT_ASSERTION_HEADER': System.getenv('ENFORCE_CLIENT_ASSERTION_HEADER'),
```
Comment (irinaepshteyn) [inline docker.gradle:105]: These changes, line 105 through 110, should be made on predix_prunes/jwt_grant_75.6.0 branch, not on rc_75.6.0 directly.

Overall, dockerization changes to dependencies.gradle, docker.gradle, uaa-docker.yml should be merged first to predix_prunes/jwt_grant_75.6.0.  Then both predix_prunes/jwt_grant_75.6.0 and predix_extensions_75.6.0 should be merged to rc_75.6.0

File: uaa/src/main/resources/uaa.yml [hunk 2]
```diff
@@ -2,10 +2,6 @@
 # if any of these exist:
 # [$UAA_CONFIG_URL, $UAA_CONFIG_PATH/uaa.yml, $CLOUDFOUNDRY_CONFIG_PATH/uaa.yml]
```
Comment (irinaepshteyn) [inline uaa/src/main/resources/uaa.yml:4]: This file should not be change with the merge. Revert this change. 

Also, this setting should be added to uaa-docker.yml to support JwtBearerGrantAT as in [uaa-docker.yml#L5-L8](https://github.com/GESoftware-CF/uaa/blob/predix_prunes/jwt_grant_75.6.0/uaa/src/main/resources/uaa-docker.yml#L5-L8) in /predix_prunes/jwt_grant_75.6.0 branch

**Review by irinaepshteyn — CHANGES_REQUESTED** [review]

File: docker.gradle (hunk 1 above)
Comment (akarollil) [reply docker.gradle:105]: That makes sense. Will do.

File: uaa/src/main/resources/uaa.yml (hunk 2 above)
Comment (akarollil) [reply uaa/src/main/resources/uaa.yml:4]: All tests pass without the device config lines (see link to passing job in comment). But yeah, there are references to this device configuration, but no tests fail. I guess there are missing tests - anyway, I will leave the lines alone.

File: docker.gradle (hunk 1 above)
Comment (akarollil) [reply docker.gradle:105]: Actually, it is messy. We seem to have done the dockerization work in predix_extensions, then somebody copied it over to the jwt_grant branch, but then all updates to dockerization have been done only for the predix_extensions branch. So jwt branch has a docker.gradle that is quite out of date with predix_extensions. And there is not 'merging' the changes from predix_extensions into jwt_grant - so we will have to copy the changes, which is wrong. So since the dockerization has to happen in a way that works across branches (predix_extensions, jwt_grant, maybe others), it might make sense to have another branch for dockerization. I guess for now, I will copy the fixes over to the jwt branch from the predix_extensions branch. So that we can build a docker container from the jwt_grant branch also even though we don't do that now.

Comment (irinaepshteyn) [reply docker.gradle:105]: I agree it's messy... we will not be able to build docker image from /predix_prunes/jwt_grant_75.6.0 as it doesn't contain necessary Dockerfile and Jenkinsfile files...  However, changes to docker.gradle and uaa-docker.yml are specific to the device authentication and should be done on the /predix_prunes/jwt_grant_xxx branch to support future merges.

**Comment by akarollil** [comment]
Dockerize pipeline build with JWT Bearer Grant Acceptance test passing for Dockerize UAA pipeline: https://i.ci.build.ge.com/bjmjek1k/ci/job/UAA/job/Dockerize/job/PR-354/3/
